### PR TITLE
U4X-360 : Add referral in the lab results on the clinician side

### PR DESCRIPTION
**Pull request by jabahum**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

##  Summary

- Referral Tab Integration:
     Added a new Referral Tab to the  investigative results with a referral tables to visualise referred orders for a patient .
- Active and Past Test Order Tables:
     Implemented two tables for Active and Past Test Orders.
- Email Results Delivery:
     Added the capability to send test results via email for  active ,past test and referral orders tables.
- Result Reporting in Referrals Tab:
     Enabled result reporting for test orders on the Referral Table in the Referrals Tab.
- Request Test Orders Feature:
    Added a feature to request a test order directly from the Active Test Order table using add button.

## Screenshots

https://github.com/openmrs/openmrs-esm-laboratory/assets/24471994/5e042149-a01f-4a39-aa57-703a3e0e39de

### CODE REVIEW

**Comment by sherrif10** [comment]
Well done on this, How i wish you split it in different tasks.

**Comment by pirupius** [comment]
@jabahum is this PR still WIP or still necessary? It has really grown (now at 33 files) 
Please add a description and screenshots/recording as well for more context

**Comment by jabahum** [comment]
> @jabahum is this PR still WIP or still necessary? It has really grown (now at 33 files) Please add a description and screenshots/recording as well for more context

@pirupius this is still necessary.

File: src/components/cards/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from "./card-header.component";
```
Comment (pirupius) [inline src/components/cards/index.ts:1]: ⛔

**Review by pirupius — COMMENTED** [review]
There PR has a bunch of changes. Please add a video recording or screenshots of how the workflow. I see some components being created within the patient-chart directory so does this mean there other widgets being added?

File: src/components/cards/card-header.component.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import classNames from "classnames";
+import { useLayoutType } from "@openmrs/esm-framework";
+import styles from "./card-header.scss";
+
+interface CardHeaderProps {
+  title: string;
+  children: React.ReactNode;
+}
+
+export function CardHeader({ title, children }: CardHeaderProps) {
```
[the remark this answers is not in the document]
Comment (jabahum) [reply src/components/cards/card-header.component.tsx]: i have been able to remove this and use the one in the commons-lib

File: src/components/cards/card-header.scss [hunk 3]
```diff
@@ -0,0 +1,43 @@
+@use '@carbon/styles/scss/spacing';
+@use '@carbon/styles/scss/type';
+@import '~@openmrs/esm-styleguide/src/vars';
+
+.desktopHeader, .tabletHeader {
```
[the remark this answers is not in the document]
Comment (jabahum) [reply src/components/cards/card-header.scss]: same applies here

File: src/components/cards/index.ts (hunk 1 above)
Comment (jabahum) [reply src/components/cards/index.ts]: this also has been removed

File: src/routes.json [hunk 4]
```diff
@@ -50,6 +50,10 @@
             "name": "results-summary",
             "component": "resultsSummaryWorkSpace"
         },
+        {
+            "name" :"",
+            "component": ""
+         },
```
Comment (pirupius) [inline src/routes.json:56]: this needs clean up

**Review by pirupius — APPROVED** [review]

**Comment by sherrif10** [comment]
Well done @jabahum  @pirupius , there are couple of fixes we need to get rid of expecially on laboratoryorder form.

**Comment by sherrif10** [comment]
Commiting a followup commit on editing referal form , its not visible when you try edditing.

**Pull request merged**